crates/daemon/tests/common/mod.rs

```diff
@@ -0,0 +1,165 @@
+use assert_cmd::prelude::*;
+use std::{
+    io::{BufRead, BufReader, Write},
+    net::TcpStream,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+    sync::OnceLock,
+    thread::{self},
+    time::{Duration, Instant},
+};
+
+struct ManagedChild {
+    child: Child,
+}
+impl ManagedChild {
+    fn new(name: &'static str, mut child: Child) -> Self {
+        // Rust tests capture output, and hide it if the test passes unless `--nocapture` is passed to `cargo test`.
+        // This does *not* automatically apply to subprocesses, so: start threads to send subprocess output through
+        // `print!` / `eprintln!` to get the same behavior.
+        let stdout = child.stdout.take().expect("Failed to get stdout");
+        let stderr = child.stderr.take().expect("Failed to get stderr");
+        thread::spawn(|| {
+            let name = name.to_string();
+            let reader = BufReader::new(stdout);
+            for line in reader.lines() {
+                println!("[{name}]: {}", line.expect("Failed to read line"));
+            }
+        });
+        thread::spawn(|| {
+            let name = name.to_string();
+            let reader = BufReader::new(stderr);
+            for line in reader.lines() {
+                eprintln!("[{name}]: {}", line.expect("Failed to read line"));
+            }
+        });
+        Self { child }
+    }
+}
+impl Drop for ManagedChild {
+    fn drop(&mut self) {
+        self.child.kill().expect("Failed to kill child process");
+    }
+}
+
+fn start_daemon(db: &Path) -> ManagedChild {
+    let mut minimal_db = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    minimal_db.push("tests/Test.db");
+
+    ManagedChild::new(
+        "daemon",
+        Command::cargo_bin("moor-daemon")
+            .expect("Failed to find moor-daemon binary")
+            .arg("--textdump")
+            .arg(minimal_db)
+            .arg("--generate-keypair")
+            .arg(db)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("Failed to start daemon"),
+    )
+}
+
+static TELNET_HOST_BIN: OnceLock<PathBuf> = OnceLock::new();
+fn telnet_host_bin() -> &'static PathBuf {
+    TELNET_HOST_BIN.get_or_init(|| {
+        escargot::CargoBuild::new()
+            .bin("moor-telnet-host")
+            .manifest_path("../telnet-host/Cargo.toml")
+            .current_release()
+            .run()
+            .expect("Failed to build moor-telnet-host")
+            .path()
+            .to_owned()
+    })
+}
+
+// Because of this function, it might be cleaner to host integration tests
+// in a separate crate.
```

Comment:
... or actually, maybe `telnet-host` would be a better place? /shrug

Reply:
Yeah, I think the `tests/` directory of `telnet-host` is the right place for that. 

For modularity/coupling reasons in fact I think there should be no references into `telnet-host` from `daemon` at all. So it'd be a matter of moving this code into `crates/telnet-host/tests` having a `dev-dependency` inside `telnet-host/Cargo.toml` referring to the daemon? 

In general that's the approach integration tests should be following for cargo project (see integration tests here) though I have personally not been consistent with that.

https://doc.rust-lang.org/book/ch11-03-test-organization.html